File: leetcode/python/566.重塑矩阵.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        m = len(mat)
        if m == 0:
            return mat
        n = len(mat[0])
        if n == 0:
            return mat
        if m * n != r * c:
            return mat
        
        rows = [0 for _ in range(c)]
        ret = [list(rows) for _ in range(r)]
        for i in range(m):
            for j in range(n):
                idx = i * n + j
                row = idx // c
                col = idx % c
                ret[row][col] = mat[i][j]
        return ret

File: leetcode/golang/566.重塑矩阵.go
func matrixReshape(mat [][]int, r int, c int) [][]int {
    n := len(mat);
    if n == 0 {
        return mat;
    }
    m := len(mat[0]);
    if m == 0 {
        return mat;
    }
    if m * n != r * c {
        return mat;
    }
    
    rr := 0;
    ret := make([][]int, r)
    for {
        if rr >= r {
            break;
        }
        rows := make([]int, c);
        ret[rr] = rows;
        rr += 1;
    }

    i := 0;
    for {
        if i >= n {
            break;
        }
        j := 0;
        for {
            if j >= m {
                break;
            }
            idx := i * m + j;

            row := idx / c;
            col := idx % c;
            ret[row][col] = mat[i][j]

            j += 1;
        }
        i += 1;
    }
    return ret;
}